client/client-lib/src/api.rs

```diff
@@ -55,6 +55,10 @@ pub trait FederationApi: Send + Sync {
     /// Fetch preimage offer for incoming lightning payments
     async fn fetch_offer(&self, payment_hash: Sha256Hash) -> Result<IncomingContractOffer>;
 
+    // TODO: find a better abstraction for all our API endpoints that allows different strategies and timeouts
+    /// Checks if there exists an offer for a payment hash
+    async fn offer_exists(&self, payment_hash: Sha256Hash) -> Result<bool>;
```

Comment:
Is the end goal for this to be a some kind of general method for all server modules, or would this be better served as a private endpoint specific to ln module?

Reply:
In a modularized future that would be an API endpoint specific to the LN module. But currently everything lives in `FederationApi`.

Reply:
```
fn api_endpoints(&self) -> &'static [ApiEndpoint<Self>]
```

What about this one on `FederationApi`?

Reply:
That's on the server side to register module endpoints with the federation's API server.